src/util/mapValues.js

```diff
@@ -2,10 +2,18 @@
  * Maps all the values in the given object through the given function and saves them, by key, to a result object
  */
 const mapValues = (obj, fn) => {
-  return obj ? Object.keys(obj).reduce((accumulator, key) => ({
-    ...accumulator,
-    [key]: fn(obj[key], key)
-  }), {}) : obj
+  if (obj !== Object(obj)) {
+    return obj
+  } else {
+    const acc = {}
```

Comment:
Previous version was extensively creating `accumulator` objects.
It's ok to mutate `acc`...